api/views/views.py

```diff
@@ -576,21 +579,30 @@ def get(self, request, path, fname, format=None):
             # or the photo is shared with the user
             image_hash = fname.split(".")[0].split("_")[0]  # janky alert
             user = User.objects.filter(id=token["user_id"]).only("id").first()
-            if photo.owner == user or user in photo.shared_to.all():
+
+            if internal_path is not None:
                 response = HttpResponse()
                 mime = magic.Magic(mime=True)
                 filename = mime.from_file(photo.main_file.path)
                 response["Content-Type"] = filename
                 response["X-Accel-Redirect"] = internal_path
+            else:
+                try:
+                    response = FileResponse(open(photo.main_file.path, 'rb'))
```

Comment:
Potential security issue ? Allowing potential read of arbitrary files ?

Reply:
Can you explain your concerns in a bit more detail? 

This part is for getting the original photo, that is on the file system. There has to be an actual photo entity with a corresponding file entity in order to read that particular path.

Usually this gets redirected to the proxy, which responds with the original file from the file system. 

However in this edge case, the internal path is not valid, which can happen, if the hosted system is not a docker setup and there is no "PHOTOS" environment variable according to the code.

Reply:
Just saw the raw open(photo.main_file.path, 'rb'), if photo.main_file_path is in any way coming from user input and is not validated, it could allow reads of unintended files, path traversals and so on.
If its validated or is not coming from user input, then no worries..

Reply:
Potentially, yeah. If users can manipulate the main_file path, then they could set it to the path of secret keys or anything readable by the librephotos user. I think the only way to set that path is via django admin or editing sql, but maybe that's too big of a risk?